Fix creation of the APT repo directory structure for similar distros

DieterReuter opened the pull request:
Right now we do have a problem to store the .debs for `raspbian-jessie` and
`debian-jessie` distro version for `armhf` arch. Both .debs have the same filename
so we have to include the distro version, too.

Signed-off-by: Dieter Reuter dieter.reuter@me.com

icecrime commented:
Cc @andrewhsu

andrewhsu commented:
@DieterReuter looks like we're already including the distro codename in the filename. I'd recommend having the filename changed instead of having another directory level introduced.

/cc @mlaventure

mlaventure commented:
@DieterReuter, @andrewhsu point makes sense.

Maybe we can update `hack/make/build-deb` instead. I think adding the distro name [here](https://github.com/docker/docker/blob/master/hack/make/build-deb#L81) would do the trick (e.g. `RUN { echo '$debSource (${debVersion}-0~${distro}-${suite}) $suite; urgency=low ...` )

DieterReuter commented:
@mlaventure agreed, this would do the trick too without changing the directory structure and without any possible conflicts at all!

DieterReuter committed:
Include distro name into .deb filename

Right now we do have a problem to store the .debs for raspbian-jessie and
debian-jessie distro version for armhf arch. Both .debs have the same filename
so we have to include the distro version, too.

Signed-off-by: Dieter Reuter <dieter.reuter@me.com>

DieterReuter commented:
@andrewhsu I've just changed the .deb filename, pretty small change. :innocent:

mlaventure commented:
@DieterReuter so I tested this, and it works correctly.

We however need a change in `hack/make/release-deb` for the new file to be correctly taken into account.

Would you mind incorporating it in your PR?

Here's the diff:
```diff
diff --git a/hack/make/release-deb b/hack/make/release-deb
index 19faae1..ad750aa 100755
--- a/hack/make/release-deb
+++ b/hack/make/release-deb
@@ -132,6 +132,7 @@ for dir in bundles/$VERSION/build-deb/*/; do
 
        # update the filelist for this codename/component
        find "$APTDIR/pool/$component" \
+               -name *~${codename}*.deb -o \
                -name *~${codename#*-}*.deb > "$APTDIR/dists/$codename/$component/filelist"
 done
```

DieterReuter commented:
@mlaventure Good catch, I just updated the PR.

mlaventure commented:
LGTM

ping @tianon @andrewhsu

andrewhsu commented:
LGTM

thaJeztah reviewed (approved):
thanks for reviewing @andrewhsu 

LGTM

DieterReuter commented:
Cool, and finally we support Debian on ARM as well. 🎉
